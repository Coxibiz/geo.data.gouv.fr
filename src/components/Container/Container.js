import React from 'react'
import PropTypes from 'prop-types'

import styles from './Container.scss'

const Container = ({ children, style }) => (
  <div className={styles.container} style={style}>
    {children}
  </div>
)

Container.propTypes = {
  children: PropTypes.node,
  style: PropTypes.object
}

export default Container
