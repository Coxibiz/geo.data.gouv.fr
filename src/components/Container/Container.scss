.container {
  position: relative;
  width: 100%;
  max-width: 1200px;
  margin: 0 auto;
  padding: 0 20px;

  @media (min-width: 400px) {
    width: 85%;
    padding: 0;
  }

  @media (min-width: 550px) {
    width: 80%;
  }
}
