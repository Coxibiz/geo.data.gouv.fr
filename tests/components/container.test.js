import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import Container from '../../components/container'

describe('<Container />', () => {
  test('should set maxWidth when not fluid', () => {
    const element = shallow(<Container />).getElement()

    expect(element.props.style.maxWidth).toBe(1200)
  })

  test('should not set maxWidth when fluid', () => {
    const element = shallow(<Container fluid />).getElement()

    expect(element.props.style.maxWidth).toBe(null)
  })

  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Container>
          Some content
        </Container>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
