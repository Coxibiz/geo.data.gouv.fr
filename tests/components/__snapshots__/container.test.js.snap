// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> render correctly 1`] = `
<div
  className="jsx-1157840685"
  style={
    Object {
      "maxWidth": 1200,
    }
  }
>
  Some content
</div>
`;
